accio-tests/src/test/java/io/accio/testing/bigquery/TestBigQueryCacheService.java

```diff
@@ -72,12 +75,18 @@ private TestBigQueryCacheService()
                 .setCredentialsKey(getenv("TEST_BIG_QUERY_CREDENTIALS_BASE64_JSON"))
                 .setProjectId(getenv("TEST_BIG_QUERY_PROJECT_ID"))
                 .setBucketName(getenv("TEST_BIG_QUERY_BUCKET_NAME"));
-        BigQueryCredentialsSupplier bigQueryCredentialsSupplier = provideBigQueryCredentialsSupplier(bigQueryConfig);
-        HeaderProvider headerProvider = createHeaderProvider();
-        BigQueryClient bigQueryClient = provideBigQuery(bigQueryConfig, headerProvider, bigQueryCredentialsSupplier);
+        BigQueryCredentialsSupplier bigQueryCredentialsSupplier = new BigQueryCredentialsSupplier(bigQueryConfig.getCredentialsKey(), bigQueryConfig.getCredentialsFile());
+        HeaderProvider headerProvider = FixedHeaderProvider.create("user-agent", "accio/1");
+        ConfigManager configManager = new ConfigManager(
```

Comment:
Could we add TestingConfigManager with builder pattern to reduce useless config?

Reply:
I'm not sure what you mean but I think we should assert other unused configurations won't be impacted when we only set the specific configuration.

Reply:
They should be not impacted each other. We don't over design them.